### PR TITLE
Gitlab cleanup job

### PR DESCRIPTION
**Pull request by mcanoy**
#### What is this PR About?
Adds a cronjob that can delete stale gitlab groups and projects. This includes a github action to build and deploy the action to quay.io. 

The job takes a group id and checks its group and sub groups for projects that are older than 'x' amount of hours. If older, then they are deleted. If all of the sub group projects are deleted then the group is also deleted. 
- The parent group can never be deleted
- The job searches for `DO_NOT_DELETE` in the group and project descriptions and tag_list. If found then the group / project will not be deleted
- Provides a dry-run capability that will not delete anything but only log out what would have been deleted (on by default)
- provides a github action to build the container for this job. Builds a container labeled with the git sha when pushing to master as well as labeling it master. 
- provides a github action to tag the container on release with the release version and also latest. 
- **requires** set of a quay.io repo for `gitlab-cleanup` with appropriate bot user / pass and access

#### How do we test this?

A couple of options:

docker locally 

```
docker build . -t gitlab-clean
docker run -e DRY_RUN=true -e PARENT_GROUP_ID=-10 -e GIT_TOKEN=xxxxx -e GITLAB_API_URL=https://gitlab.com -e DELETE_AFTER_HOURS=240000 -e LOG_LEVEL=DEBUG gitlab-clean
```

from my quay:

Set a group and projects to play with (not deletes by default). Check the Readme here for full config options.

```
helm template . \
  --set env.secret.name=my-openshift-secret-ref \
  --set env.secret.gitlabApiUrl=https//my.gitlab.base.com \
  --set env.secret.personalAccessToken=bot-token-value \
  --set env.secret.parentRepoId=-1 \
| oc apply -f - -n target-namespace
```

cc: @redhat-cop/casl, @oybed, @pabrahamsson

### CODE REVIEW

File: jobs/gitlab-delete-stale-projects/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
```
Comment (tylerauerbeck) [inline jobs/gitlab-delete-stale-projects/Chart.yaml]: @mcanoy Can we bump this to v2 for Helm 3?

**Review by tylerauerbeck — CHANGES_REQUESTED** [review]
LGTM. One small fix and this should be good to go. May need to look into what we need to do to "enable" the github actions here, but I can dig into that seperately.

File: jobs/gitlab-delete-stale-projects/Chart.yaml (hunk 1 above)
Comment (mcanoy) [reply jobs/gitlab-delete-stale-projects/Chart.yaml]: ack v2 for H3. updated.

**Comment by pabrahamsson** [comment]
@tylerauerbeck @mcanoy I've created the Quay repo: [https://quay.io/redhat-cop/gitlab-cleanup](https://quay.io/repository/redhat-cop/gitlab-cleanup) and also the `QUAY_USERNAME` & `QUAY_PASSWORD` secrets

**Review by tylerauerbeck — APPROVED** [review]
LGTM

**Pull request merged**